azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/sync_clients.py

```diff
@@ -90,68 +157,20 @@ def callback():
         self._transport.send_event(message, callback)
         send_complete.wait()
 
-    def _emit_connection_status(self):
-        """
-        The connection status is emitted whenever the client on the module gets connected or disconnected.
-        """
-        logger.info("emit_connection_status: {}".format(self.state))
-        if self.on_connection_state:
-            self.on_connection_state(self.state)
-        else:
-            logger.warn("No callback defined for sending state")
 
-    def _handle_transport_connected_state(self, new_state):
-        self.state = new_state
-        self._emit_connection_status()
-
-    def _handle_transport_event_sent(self):
-        logger.info("_handle_transport_event_sent: " + str(self.on_event_sent))
-        if self.on_event_sent:
-            self.on_event_sent()
-
-    @classmethod
-    def from_authentication_provider(cls, authentication_provider, transport_name):
-        """
-        Creates a device client with the specified authentication provider and transport
-
-        When creating the client, you need to pass in an authorization provider and a transport_name
-
-        The authentication_provider parameter is an object created using the authentication_provider_factory
-        module.  It knows where to connect (a network address), how to authenticate with the service
-        (a set of credentials), and, if necessary, the protocol gateway to use when communicating
-        with the service.
-
-        The transport_name is a string which defines the name of the transport to use when connecting
-        with the service or the protocol gateway.
+class DeviceClientSync(GenericSyncClient):
```

Comment:
How do you guys feel about this naming?

Having the Sync in different places in the generic, and the user facing version annoys me, but GenericSyncClient sounds better than GenericClientSync (especially given the contrast with GenericAsyncClient), while DeviceClientSync sounds better than DeviceSyncClient (because it has to contrast with DeviceClient)

Open to suggestion here.

Reply:
I strongly believe in consistency here, and I agree that it sounds better as GenericSyncClient, but I always regret going for "It sounds like better English, so I'm going to break the pattern here".  
If it makes the little high-school English teacher that stands on your shoulder happy, you can think of Sync as the same as Synchronous, and if you call it SyncGenericClient, it really means SynchronousGenericClient and that's two adjectives modifying a noun. (I don't like having mine up there, but she gives better advice than both the devil on my left shoulder and the angel on my right shoulder.)

---
In reply to: [246597321](https://github.com/Azure/azure-iot-sdk-python-preview/pull/39#discussion_r246597321) [](ancestors = 246597321)

Reply:
I would prefer having it as `GenericClientSync` and `DeviceClientSync`..  

---
In reply to: [246951779](https://github.com/Azure/azure-iot-sdk-python-preview/pull/39#discussion_r246951779) [](ancestors = 246951779,246597321)